SitumWayfinding/Classes/Positioning/PositioningViewController.swift

```diff
@@ -1308,4 +1316,63 @@ extension PositioningViewController {
     }
 }
 
-
+extension PositioningViewController {
+    func initializeFloorChangerMarker() {
+        let icon = self.scaledImage(
+            image: UIImage(named: "change_floor")!,
+            scaledToSize: CGSize(width: 40.0, height: 40.0)
+        )
+        let markerView = UIImageView(image: icon)
+        self.changeOfFloorMarker.iconView = markerView
+        self.changeOfFloorMarker.position = CLLocationCoordinate2D(latitude: 0, longitude: 0)
+        resetFloorMarker()
+    }
+    
+    func showFloorChangeMarker(position: CLLocationCoordinate2D) {
+        self.changeOfFloorMarker.position = position
+        self.changeOfFloorMarker.map = self.mapView
+    }
+    
+    func scaledImage(image:UIImage, scaledToSize newSize:CGSize) -> UIImage{
+        UIGraphicsBeginImageContextWithOptions(newSize, false, 0.0)
+        image.draw(in: CGRect(x: 0, y: 0, width: newSize.width, height: newSize.height))
+        let newImage:UIImage = UIGraphicsGetImageFromCurrentImageContext()!
+        UIGraphicsEndImageContext()
+        return newImage
+    }
+    
+    func changePositionFloorMarkerOrResetFloorMarker(segments: Array<SITRouteSegment>, selectedFloor: SITFloor) {
+        resetFloorMarker()
+        
+        if segments.count > 1 {
+            for segment in segments {
+                if isNotLastSelectedFloorDestiny(segment: segment, selectedFloor: selectedFloor) {
+                 
+                    let last = segment.points.last
+                    if let coordinates = last?.coordinate() {
+                        let position = CLLocationCoordinate2D(
+                            latitude: coordinates.latitude,
+                            longitude: coordinates.longitude
+                        )
+                        self.showFloorChangeMarker(position: position)
+                    }
+                }
+                
+                if self.lastSelectedMarker?.floorIdentifier == selectedFloor.identifier  {
+                    resetFloorMarker()
+                }
+            }
+        } else {
+            resetFloorMarker()
+        }
+    }
+    
+    func isNotLastSelectedFloorDestiny(segment: SITRouteSegment, selectedFloor: SITFloor) -> Bool {
```

Comment:
Nesta funcion estas metendo duas cousas distintas
1. Comprobar se o segemento está no floor que se vai pintar pintar (is)
2. Comprobar se o floor que se vai pintar é o de destino 
Se o metes todo xunto queda moi confuso, e o nome tampouco axuda